app/code/Magento/Customer/Model/ResourceModel/GroupExcludedWebsiteRepository.php

```diff
@@ -53,7 +56,7 @@ public function save(GroupExcludedWebsiteInterface $groupExcludedWebsite): Group
     public function getCustomerGroupExcludedWebsites(int $customerGroupId): array
     {
         try {
-            return $this->groupExcludedWebsiteResourceModel->loadCustomerGroupExcludedWebsites($customerGroupId);
+            return $this->excludedWebsitesForGroups[$customerGroupId] ??= $this->groupExcludedWebsiteResourceModel->loadCustomerGroupExcludedWebsites($customerGroupId);
```

Comment:
https://developer.adobe.com/commerce/php/coding-standards/technical-guidelines/
2.9. Service classes (ones that provide behavior but not data, like EventManager) SHOULD NOT have a mutable state.

Also you can implement as workaround Magento\Framework\ObjectManager\ResetAfterRequestInterface

Reply:
but it doesn't have a mutable state... Because it returns an array with copied data every time, which is immutable..

We could argue that we reset the data of a save is triggered which makes it the same behavior as for instance `Magento\Catalog\Api\ProductRepositoryInterface` where a save/delete also removes the object from the internal cache.

We could also move this to the `\Magento\Customer\Model\Plugin\GetByIdCustomerGroupExcludedWebsite::afterGetById`

this is what triggers this, would that be more suitable?

let me change it so we can discuss :)

Reply:
I just thought it would be more convenient to save it at the source.